src/os.zig

```diff
@@ -0,0 +1,371 @@
+//! OS-level functionality.
+//! Designed to be slightly higher level than sys.zig
+
+/// Opaque type representing the user ID.
+pub const Uid = struct {
+    const Self = @This();
+
+    _underlying: std.c.uid_t,
+
+    pub fn queryEffective() Self {
+        // TODO(markovejnovic): Unfortunately, geteuid is not available in Zig's stdlib at the
+        //                      moment, so we need to use @cImport above.
+        return .{ ._underlying = bun.c.geteuid() };
+    }
+
+    pub fn queryReal() Self {
+        // TODO(markovejnovic): Unfortunately, getuid is not available in Zig's stdlib at the
+        //                      moment, so we need to use @cImport above.
+        return .{ ._underlying = bun.c.getuid() };
+    }
+
+    /// Falls back to queryEffective.
+    pub fn queryCurrent() Self {
+        return queryEffective();
+    }
+};
+
+/// Utilities for fetching and interacting with the current system home directory.
+pub const HomeDir = struct {
+    /// Deduces the current user's home directory.
+    ///
+    /// Has multiple strategies depending on the platform and environment. Mostly compliant with
+    /// the POSIX standard for POSIX systems.
+    ///
+    /// No compliance claims are made for Windows.
+    ///
+    /// There is actually a reasonable desire to be able to query other users' home directories.
+    /// However, we don't have a need to do that at this moment. See
+    /// doi:10.1109/IEEESTD.2018.8277153 2.6.1 for further details.
+    pub fn query(allocator: std.mem.Allocator) !bun.Environment.OsTypeSelect(.{
```

Comment:
Would be good to specify the error set explicitly since this code uses some custom errors like `FailedToFindHomeDir`.